akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -131,12 +132,17 @@ object HttpEntity {
   def apply(contentType: ContentType, contentLength: Long, data: Source[ByteString, Any]): UniversalEntity =
     if (contentLength == 0) empty(contentType) else Default(contentType, contentLength, data)
 
-  def apply(contentType: ContentType, file: File): UniversalEntity = {
+  def apply(contentType: ContentType, file: File, chunkSize: Int = SynchronousFileSource.DefaultChunkSize): UniversalEntity = {
     val fileLength = file.length
-    if (fileLength > 0) Default(contentType, fileLength, ???) // FIXME: attach from-file-Publisher
+    if (fileLength > 0) Default(contentType, fileLength, SynchronousFileSource(file, chunkSize))
     else empty(contentType)
   }
 
+  /**
+   * Java API
+   */
+  def create(contentType: ContentType, file: File): UniversalEntity = apply(contentType, file)
```

Comment:
I see, because of the default value. 
It would be good to not expose the `create` to users here (or expose it "fully" (two methods)), consistency is dog :)